src/inference/src/dev/core_impl.cpp

```diff
@@ -57,8 +57,251 @@ void stripDeviceName(std::string& device, const std::string& substr) {
+    for (auto property = result_properties.begin(); property != result_properties.end();) {
+        if (property->first != ov::device::properties.name()) {
+            // 1. Skip non-matching properties
+            property++;
+            continue;
+        }
+
+        // 2. device properties DEVICE_PROPERTIES are found
+        if (property->second.is<std::string>()) {  // because of legacy API 1.0
+            property->second = property->second.as<ov::AnyMap>();
+        }
+        auto& secondary_properties = property->second.as<ov::AnyMap>();
+
+        for (auto secondary_property = secondary_properties.begin();
+             secondary_property != secondary_properties.end();) {
+            // flattening is performed only when config is applicable (see docs for ov::is_config_applicable)
+            if (ov::is_config_applicable(user_device_name, secondary_property->first)) {
+                // 2.1. flatten the secondary property for target device
+                // example: core.compile_model("GPU", ov::device::properties("GPU", ov::prop1));
+                // example: core.compile_model("GPU.1", ov::device::properties("GPU", ov::prop1));
+                update_result_properties(secondary_property->second.as<ov::AnyMap>());
+                secondary_property = secondary_properties.erase(secondary_property);
+            } else if (is_virtual_device(user_device_name)) {
+                // 2.2. keep the secondary property for the other virtual devices
+                secondary_property++;
+                continue;
+            } else {
+                // 2.3. remove the secondary property setting for other hardware device
+                // example: core.compile_model("GPU", ov::device::properties("CPU", ov::prop1));
+                secondary_property = secondary_properties.erase(secondary_property);
+            }
+        }
+
+        // 3. go to the next property
+        if (secondary_properties.empty()) {
+            // 3.1. since the sub-property is flattened, we need to drop it
+            property = result_properties.erase(property);
+        } else {
+            // 3.2. some properties are still in ov::device::properties(ov::AnyMap{}), abort loop
+            break;
+        }
+    }
+
+    return result_properties;
+}
+
+enum class MatchType { EXACT = 0, SUBSTR };
+
+struct DevicePriority {
+    std::string prop_name;
+    MatchType match_type;
+};
+
+DevicePriority get_device_priority_property(const std::string& device_name) {
+    return is_virtual_device(device_name)
+               ? DevicePriority{ov::device::priorities.name(), MatchType::EXACT}
+               :
+               // ov::device::properties(GPU.0) can be applied for GPU tile identified by GPU.0.0
+               DevicePriority{ov::device::id.name(), MatchType::SUBSTR};
+}
+
+void clean_batch_properties(std::string& deviceName, ov::AnyMap& config, const ov::PropertyName& property_name) {
```

Comment:
```suggestion
void clean_batch_properties(const std::string& deviceName, ov::AnyMap& config, const ov::PropertyName& property_name) {
```

Reply:
will be fixed in current PRs if we have more major comments. 
Otherwise, suggest fixing in another PR

Reply:
fixed in current PR